### PR TITLE
Labeling excess median

### PR DESCRIPTION
**Pull request by ruifanp**
Jupyter Notebook for labelling excess over median method.

### CODE REVIEW

File: Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb [hunk 1]
```diff
@@ -0,0 +1,678 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
```
Comment (PanPip) [inline Labelling/Labels Fixed Horizon/Fixed Time Horizon.ipynb:7]: Please remove Fixed Time Horizon Notebook from this PR, as it's available in [PR#142](https://github.com/hudson-and-thames/research/pull/142)

File: Labelling/Labels Excess Over Median/Excess over Median.ipynb [hunk 2]
```diff
@@ -0,0 +1,1000 @@
...
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Labeling: Excess Return Over Median\n",
+    "\n",
+    "![image_example](img/distribution_over_median.png)\n",
+    "_*Fig. 1:*_ Distribution of excess over median return for 22 stock tickers from period between Jan 2019 and May 2020.\n",
+    "\n",
```
Comment (PanPip) [inline Labelling/Labels Excess Over Median/Excess over Median.ipynb:11]: Try to use a bar plot here for the range [-0.2; 0.2] so the distribution plot of labels is more informative

File: Labelling/Labels Excess Over Median/Excess over Median.ipynb [hunk 3]
```diff
@@ -0,0 +1,1000 @@
...
+    "## Abstract\n",
+    "\n",
+    "In this notebook, we demonstrate the method of labelling financial returns data according to excess over median. Using cross sectional data on returns of many different stocks, each observation is labelled according to whether or how much its return exceeds the median. Correlations can then be found between features and the likelihood that a stock will outperform the market.\n",
+    "\n",
```
Comment (PanPip) [inline Labelling/Labels Excess Over Median/Excess over Median.ipynb:15]: `labelling` to `labeling`
`cross sectional` to `cross-sectional`
`labelled` to `labeled`
for consistency

File: Labelling/Labels Excess Over Median/Excess over Median.ipynb [hunk 4]
```diff
@@ -0,0 +1,1000 @@
...
+    "This technique is used in the following paper:\n",
+    "[\"The benefits of tree-based models for stock selection\"](https://link.springer.com/article/10.1057/jam.2012.17) by _Zhu et al._ (2012). \n",
+    "\n",
+    "In that paper, independent composite features are constructed as weighted averages of various parameters in fundemental and quantitative analysis, such as PE ratio, corporate cash flows, debt etc. The composite features are the applied as parameters in linear regression or a decision tree to predict whether a stock will outperform the market, as defined by excess over median.\n",
+    "\n",
```
Comment (PanPip) [inline Labelling/Labels Excess Over Median/Excess over Median.ipynb:20]: `fundemental` to `fundamental`
`features are the applied` to `features are applied`

File: Labelling/Labels Excess Over Median/Excess over Median.ipynb [hunk 5]
```diff
@@ -0,0 +1,1000 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[*********************100%***********************]  22 of 22 completed\n"
```
Comment (PanPip) [inline Labelling/Labels Excess Over Median/Excess over Median.ipynb:54]: Perhaps worth outputting head of only 8-9 stocks so the output table fits in the cell.

File: Labelling/Labels Excess Over Median/Excess over Median.ipynb [hunk 6]
```diff
@@ -0,0 +1,1000 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook presents a method to label data according to excess return over median. This method can return either numerical or categorical labels for observations. Zhu et al. utilize these labels to predict monthly stock returns using both linear regression and decision trees on composite features. In this process:\n",
+    " - Forward rates of return for assets are calculated for the entire selection of stocks indexed by time bars.\n",
+    " - At each time index, the median rate of return for all stocks is calculated. The median is subtracted from each stock's return to find the excess return over median.\n",
+    " - If desired, the excess returns are converted to categorical signs\n",
+    "\n",
+    "This method is useful for labelling data used for training regression models and decision trees. Zhu et al. found that decision trees were slightly better at predicting outperformers than linear regression."
```
Comment (PanPip) [inline Labelling/Labels Excess Over Median/Excess over Median.ipynb:961]: `labelling` to `labeling`

File: Labelling/Labels Excess Over Median/Excess over Median.ipynb (hunk 5 above)
Comment (ruifanp) [reply Labelling/Labels Excess Over Median/Excess over Median.ipynb]: How necessary is this? Personally I feel it's a bit weird if I write that there are 22 stocks but only show 8-9 of them.

Comment (PanPip) [reply Labelling/Labels Excess Over Median/Excess over Median.ipynb:75]: Fitting the table on one page would improve readability. You can output a list of stocks that you're using, however, it's up to you 🙂

**Pull request merged**